jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/MetaInfConfiguration.java

```diff
@@ -462,7 +428,9 @@ private static boolean isEmptyResource(Resource resourcesDir)
      */
     public void scanForFragment(WebAppContext context, Resource jar, ConcurrentHashMap<Resource, Resource> cache)
     {
-        Resource webFrag = null;
+        ResourceFactory resourceFactory = ResourceFactory.of(context);
```

Comment:
You should also check that `jar` is a directory now that you ensure jars are opened, shouldn't you? And specify it in the javadoc too?